src/core/include/openvino/core/type/f8e5m2.hpp

```diff
@@ -0,0 +1,167 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cmath>
+#include <iostream>
+#include <limits>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "openvino/core/core_visibility.hpp"
+#include "openvino/core/type/float16.hpp"
+
+#define ROUND_MODE_TO_NEAREST_EVEN
+
+namespace ov {
+class OPENVINO_API f8e5m2 {
```

Comment:
BTW, do we need to create similar class for nf4 ?

Reply:
That is great that we created it here but I don't think that it is required at that point. What is needed really is the way to Const_fp32/fp16 => Const_fp8 and Const_fp8 => Const_fp32/fp16. All the math happens in FP32 or FP16 data types.